### PR TITLE
feat(import): add runtime and memory import subcommands with TUI wizard

### PR DESCRIPTION
**Pull request by jesseturner21**
## Description

Adds `agentcore import runtime` and `agentcore import memory` subcommands that import existing AWS BedrockAgentCore resources into a local CLI project via CloudFormation IMPORT change sets. Also adds an interactive TUI wizard for the import flow.

### Key changes

**New import subcommands:**
- `agentcore import runtime --arn <arn> --code <path>` — imports an existing AgentCore Runtime, copies agent source code, and brings the resource under CloudFormation management
- `agentcore import memory --arn <arn>` — imports an existing AgentCore Memory with full strategy/namespace/tag preservation
- Both commands support `--name` for local name override, `--target` for multi-target projects, and `-y` for auto-confirm

**Interactive TUI wizard:**
- `agentcore import` with no args launches a guided wizard (resource type selection → ARN input → code path → progress → next steps)
- Next-steps navigation works from both the full TUI and CLI-inline contexts (deploy, status)

**Shared CDK import pipeline (`import-pipeline.ts`):**
- Extracted the duplicated CDK build → synth → bootstrap → publish → phase1 → phase2 → update-state pipeline from `actions.ts`, `import-runtime.ts`, and `import-memory.ts` into a single `executeCdkImportPipeline()` function
- Each caller provides resource-specific logic via a `buildResourcesToImport` callback
- ~200 lines of duplication removed

**AWS control plane helpers (`agentcore-control.ts`):**
- `getAgentRuntimeDetail` / `getMemoryDetail` — fetch full resource details with field validation
- `listAllAgentRuntimes` / `listAllMemories` — paginated listing with auto-select for single results
- Required field validation with clear error messages

**Import utilities (`import-utils.ts`):**
- `resolveImportContext()` — shared setup (logger, project context, target resolution)
- `parseAndValidateArn()` — validates format, resource type, region, and account match
- `failResult()` — standardized error return builder
- `copyAgentSource()` — copies agent code with exclusion filters, pyproject.toml fixup, Dockerfile generation
- `findResourceInDeployedState()` — duplicate detection against deployed-state.json

**Validation and error handling:**
- Early name validation against `AgentNameSchema` regex before any file I/O (prevents path traversal via `--name`)
- ARN format, region, and account validation before API calls
- Duplicate detection by both local name and resource ID
- Rollback on failure (config snapshot restore + copied directory cleanup)
- Auto-create deployment target from ARN when no targets exist

**Schema changes:**
- `runtimeVersion` made optional (Container builds don't need it)
- Added `description` and `encryptionKeyArn` to Memory schema
- Added `executionRoleArn` to Memory schema

## Related Issue

Closes #

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe):

## Testing

How have you tested the change?

- [x] I ran `npm run test:unit` and `npm run test:integ`
- [x] I ran `npm run typecheck`
- [x] I ran `npm run lint`
- [ ] If I modified `src/assets/`, I ran `npm run test:update-snapshots` and committed the updated snapshots

**Unit tests added:**
- `import-runtime-handler.test.ts` — 200+ tests covering entrypoint detection, field mapping, validation, duplicate detection, rollback, error paths
- `import-runtime-entrypoint.test.ts` — extractEntrypoint edge cases (OTel wrappers, missing extensions, path prefixes)
- `agentcore-control.test.ts` — AWS API helper tests (pagination, field extraction, auto-select, env var mapping)
- 260/260 import tests pass, full suite passes

**End-to-end bugbash testing:**
- Tested import memory with real AWS memory (3 strategies, tags, executionRoleArn) — all fields preserved
- Tested import runtime with real AWS CodeZip runtime — build, entrypoint, protocol all correct
- Tested error paths: invalid ARN, account mismatch, duplicate detection, missing --code
- Tested runtime + memory coexistence in same project

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [ ] I have updated the documentation accordingly
- [ ] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

File: src/cli/aws/agentcore-control.ts [hunk 1]
```diff
@@ -43,6 +47,369 @@ export async function getAgentRuntimeStatus(options: GetAgentRuntimeStatusOption
...
+export interface ListAgentRuntimesResult {
+  runtimes: AgentRuntimeSummary[];
+  nextToken?: string;
+}
+
+/**
+ * List all AgentCore Runtimes in the given region.
+ */
+export async function listAgentRuntimes(options: ListAgentRuntimesOptions): Promise<ListAgentRuntimesResult> {
+  const client = new BedrockAgentCoreControlClient({
```
[the remark this answers is not in the document]
Comment (jesseturner21) [reply src/cli/aws/agentcore-control.ts]: yes adding

File: src/cli/commands/import/import-memory.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+      if (memories.length === 0) {
+        return failResult(logger, 'No memories found in your account.', 'memory', '');
+      }
+
+      if (memories.length === 1) {
+        memoryId = memories[0]!.memoryId;
+        onProgress(`Found 1 memory: ${memoryId}. Auto-selecting.`);
+      } else {
+        console.log(`\nFound ${memories.length} memory(ies):\n`);
```
[the remark this answers is not in the document]
Comment (jesseturner21) [reply src/cli/commands/import/import-memory.ts]: lol yeah that is wack

File: src/cli/commands/import/import-memory.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    SUMMARY_OVERRIDE: 'SUMMARIZATION',
+    USER_PREFERENCE_OVERRIDE: 'USER_PREFERENCE',
+    EPISODIC_OVERRIDE: 'EPISODIC',
+    // Direct mappings
+    SEMANTIC: 'SEMANTIC',
+    SUMMARIZATION: 'SUMMARIZATION',
+    USER_PREFERENCE: 'USER_PREFERENCE',
+    EPISODIC: 'EPISODIC',
+  };
+  return mapping[apiType] ?? apiType;
```
[the remark this answers is not in the document]
Comment (jesseturner21) [reply src/cli/commands/import/import-memory.ts]: Intentional — if the API adds a new strategy type we haven't mapped yet, falling back to the raw key lets the import still work rather than hard-failing. The value flows into agentcore.json where the user can see and fix it. Could add a warning log if the type is unmapped though.
